offchain/danaswap-browser/src/DanaSwap/Browser/Home.purs

```diff
@@ -19,18 +24,239 @@ main = HA.runHalogenAff do
     Nothing -> throwError $ error "Cannot find #home"
     Just elem -> void $ runUI component unit elem
 
+type State =
+  { pools :: Array Pool
+  , currentPool :: Maybe Pool
+  , activity :: Activity
+  }
+
+data Action
+  = SetCurrentPool Pool
+  | ResetCurrentPool
+  | SetActivity Activity
+
+data Activity
+  = Swap
+  | AddLiquidity
+  | WithdrawLiquidity
+
+derive instance eqActivity :: Eq Activity
+
 component
   :: forall q i o m
    . MonadAff m
   => H.Component q i o m
 component =
   H.mkComponent
-    { initialState: const unit
+    { initialState: const
+        { pools:
+            [ Pool { assetA: Asset { name: "ADA", value: fromInt 1200 }, assetB: Asset { name: "DANA", value: fromInt 1000 }, fee: 4 }
+            , Pool { assetA: Asset { name: "ADA", value: fromInt 10000 }, assetB: Asset { name: "USDC", value: fromInt 4000 }, fee: 4 }
+            , Pool { assetA: Asset { name: "ADA", value: fromInt 120 }, assetB: Asset { name: "HOSKY", value: fromMaybe (fromInt 0) (fromString "3000000100") }, fee: 4 }
+            , Pool { assetA: Asset { name: "DADA", value: fromInt 3000 }, assetB: Asset { name: "USDC", value: fromInt 1000 }, fee: 4 }
+            ]
+        , currentPool: Nothing
+        , activity: Swap
+        }
     , render
     , eval: H.mkEval $ H.defaultEval
+        { handleAction = handleAction
+        }
     }
   where
-  render _ =
-    HH.div_
-      [ HH.text "this is Home"
+  handleAction :: forall slots. Action -> H.HalogenM State Action slots o m Unit
+  handleAction = case _ of
+    ResetCurrentPool -> H.modify_ _ { currentPool = Nothing }
+    SetCurrentPool pool -> H.modify_ _ { currentPool = Just pool }
+    SetActivity activity -> H.modify_ _ { activity = activity }
+
+  render :: forall slots. State -> H.ComponentHTML Action slots m
+  render { pools, currentPool, activity } =
+    HH.section
+      [ mkClass "section my-6 pb-6" ]
+      $
+        [ HH.div
+            [ mkClass "columns is-mobile" ]
```

Comment:
Is this really "mobile" or just a small window?

Reply:
this is all about responsiveness so the layout will not mess up with smaller windows https://bulma.io/documentation/columns/responsiveness/#mobile-columns

Reply:
I'm saying I disagree with Bulma's naming scheme. I could buy a giant 15" Android tablet and that's a "mobile" device that might prefer to see these columns. I can also my browser on my laptop in a tiling window manager split and it's still "desktop". This wording communicates the wrong thing. This is about a "narrow viewport", not the type of device being used and the words should communicate this accurately. This always seems to lead to some bad design decisions as well like assuming with a class like `isMobile` that we don't have a mouse, but it could just be my browser with the dev tools open to the right side (which is usually how I do it). It's simplistic and we have better words and queries for this type of stuff.

Reply:
that's true ...